portal-kernel/src/com/liferay/portal/kernel/settings/GlobalServiceSettingsLocator.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.kernel.settings;
+
+/**
+ * @author Drew Brokke
+ */
+public class GlobalServiceSettingsLocator implements SettingsLocator {
+
+	public GlobalServiceSettingsLocator(String configurationPid) {
+		_configurationPid = configurationPid;
+	}
+
+	@Override
+	public Settings getSettings() throws SettingsException {
+		Settings configurationBeanSettings =
+			_settingsLocatorHelper.getConfigurationBeanSettings(
+				_configurationPid);
+
+		return _settingsLocatorHelper.getGlobalPortalPreferenceSettings(
+			configurationBeanSettings);
+	}
```

Comment:
Yeah, it should be enough. We should not use the preferences table for System wide configuration.